android-components/components/feature/pwa/src/main/java/mozilla/components/feature/pwa/feature/WebAppContentFeature.kt

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa.feature
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.manifest.WebAppManifest
+
+/**
+ * Feature used to handle web content settings from manifest file.
+ */
+class WebAppContentFeature(
+    private val store: BrowserStore,
+    private val tabId: String? = null,
+    private val manifest: WebAppManifest,
+) : DefaultLifecycleObserver {
+
+    override fun onCreate(owner: LifecycleOwner) {
```

Comment:
Is `onCreate` the right place for this? My guess is probably yes since we want to keep the that display mode from creation - the value doesn't change until we re-fetch a new manifest.

This would not affect the entry point from the main application in Fenix because we add this to the ExternalAppBrowserFragment.